### PR TITLE
added support for parallel instances

### PR DESCRIPTION
**Pull request by aviatam**
### Description :notebook_with_decorative_cover:

added support for xml creation with supporting parallel instances format 

### Change Classification :books:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

### Dependencies :stop_sign:

Please mention any dependencies relevant to this code change (e.g. other pull requests).

### CODE REVIEW

**Review by yoavgeva — APPROVED** [review]

File: src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java [hunk 1]
```diff
@@ -37,27 +37,54 @@ public void generate() {
...
+
+					});
+					XmlTest xmlTest = new XmlTest(topLevelSuite);
+					xmlTest.setName(getSuiteName());
+					if (getTestLevelTimeoutInMilliseconds() > 0) {
+						xmlTest.setTimeOut(getTestLevelTimeoutInMilliseconds());
+					}
+					xmlTest.setXmlClasses(classes);
+					topLevelTestsList.add(xmlTest);
+				default:
```
Comment (yoavgeva) [inline src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java:85]: throw error here

**Review by yoavgeva — CHANGES_REQUESTED** [review]

File: src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java (hunk 1 above)
Comment (aviatam) [reply src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java:85]: @yoavgeva fixed please review again

File: src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java [hunk 2]
```diff
@@ -37,27 +37,54 @@ public void generate() {
...
+							if (getTestLevelTimeoutInMilliseconds() > 0) {
+								xmlTest.setTimeOut(getTestLevelTimeoutInMilliseconds());
+							}
+							xmlTest.setXmlClasses(classes);
+							topLevelTestsList.add(xmlTest);
+						});
+					});
+				case "classes":
+				case "instances":
+					methodsPerClassMap.forEach((cls, methods) -> {
```
Comment (yoavgeva) [inline src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java:66]: enter the logic inside private method

File: src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java [hunk 3]
```diff
@@ -37,27 +37,54 @@ public void generate() {
...
-				xmlTest.setXmlClasses(classes);
-				topLevelTestsList.add(xmlTest);
-			});
-		});
+		List<XmlClass> classes = new ArrayList<>();
+		String parallelMode = getParallelMode();
+			switch (parallelMode) {
+				case "tests":
+				case "methods":
+					methodsPerClassMap.forEach((cls, methods) -> {
```
Comment (yoavgeva) [inline src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java:45]: enter the logic inside private method

**Review by yoavgeva — CHANGES_REQUESTED** [review]

File: src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java [hunk 4]
```diff
@@ -37,7 +37,47 @@ public void generate() {
...
+				case "classes":
+				case "instances":
+					generateXmlTestParamPerClass(methodsPerClassMap);
+					break;
+				default:
+					throw new RuntimeException(String.format(" %s parallel mode is not supported , please change",parallelMode));
+			}
+	}
+
+	private void generateXmlTestParamPerClass(Map<Class<?>, List<Method>> methodsPerClassMap) {
```
Comment (aviatam) [inline src/main/java/com/salesforce/goals/TestngSuiteGoalTestsMojo.java:56]: @yoavgeva please review again

**Pull request merged**